pkg/kubelet/kubelet_volumes.go

```diff
@@ -113,6 +113,11 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
 		if allPods.Has(string(uid)) {
 			continue
 		}
+		// if the pod is within termination grace period, we shouldn't cleanup the underlying volumes
+		if kl.podKiller.IsPodPendingTerminationByUID(uid) {
+			klog.V(3).InfoS("Pod is pending termination", "podUID", uid)
```

Comment:
Will this print so often that it would require some throttling?

Reply:
Not at all... Default log level is 2, and even when log level is set to 3 this function is only called every 10 seconds and not for all pods; just terminating ones.